software/prokka/main.nf

```diff
@@ -0,0 +1,54 @@
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process PROKKA {
+    tag "$meta.id"
+    label 'prokka'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::prokka=1.14.6" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/prokka:1.14.6--pl526_0"
+    } else {
+        container "quay.io/biocontainers/prokka:1.14.6--pl526_0"
+    }
+
+    input:
+    tuple val(meta), path(fasta)
+    path proteins
+    val use_proteins
```

Comment:
Is the separate `use_proteins` required? Can't we just use the proteins file if it is specified and not otherwise?

Reply:
Not really, I was following QUAST's module for optional inputs.

Reply:
pinging @drpatelh to make sure we are following the best practices here.

Reply:
Yeah, this is one of the solutions 👍🏽 The other is to use an empty list if optional like I suggested earlier. Could maybe just use the latter and if things break then we can update.